### ISSUE TITLE
Error CS0533 is not returned when both members are internal

### ISSUE THREAD

**Issue by jonathanbyrne** [dotnet/roslyn#16591]
I'm pretty sure compiler error CS0533 should be generated for both members in the super class.  If the members are protected or public, the compiler errors are generated.

As I would expect, it seems impossible to create a concrete class that extends InvalidSuperClass without generating compiler errors.

I'm building the two classes below in Visual Studio 2017 RC in a .NET Standard 1.6 project.  

```
public abstract class InvalidBaseClass
{
    internal abstract int A { get; set; }
    internal abstract int B();
}

public abstract class InvalidSuperClass : InvalidBaseClass
{
    internal abstract int A { get; set; }
    internal abstract int B();
}
```

**Comment by gafter** [comment]
This situation gets CS0114, not CS0533.

It is true that it may be impossible to implement a hidden abstract method in the base class. My doctor tells me, if it hurts don't do that. Therefore, don't hide abstract members.

**Comment by gafter** [comment]
CS0114 is warning you about a problem you may be placing upon external clients of your API. But if the method is `internal`, there are no external clients.

**Comment by julealgon** [comment]
> It is true that it may be impossible to implement a hidden abstract method in the base class. My doctor tells me, if it hurts don't do that. Therefore, don't hide abstract members.

@gafter , what if the abstract method has been implemented in a sealed way?

I ran into a situation today that got me wondering:

```csharp
public abstract class Something
{
    protected abstract void DoStuff();
}

public abstract class MySomething : Something
{
    public new abstract void DoStuff();

    protected sealed override void DoStuff() 
    { 
    }
}
```

This raises CS0533 for me. Does that make sense if what I'm hiding is a sealed override?
